packages/repository-json-schema/src/filter-json-schema.ts

```diff
@@ -197,20 +197,38 @@ export function getFieldsJsonSchemaFor(
   modelCtor: typeof Model,
   options: FilterSchemaOptions = {},
 ): JsonSchema {
-  const schema: JsonSchema = {
-    ...(options.setTitle !== false && {
-      title: `${modelCtor.modelName}.Fields`,
-    }),
+  const objTitle =
+    options.setTitle !== false
+      ? {
+          title: `${modelCtor.modelName}.Fields`,
+        }
+      : {};
```

Comment:
I find this bit rather complex, how about the following?

```suggestion
  const objTitle =
    options.setTitle !== false
      ? `${modelCtor.modelName}.Fields`
      : undefined;
```

And later:

```ts
return {
  title: objTitle,
  oneOf
};
```

It slightly changes the return value - the current version returns object with no `title` property while the new version will return `title` set to `undefined`, I don't know if that is a problem.

Alternatively, may we can keep the current code as is to avoid unnecessary code churn?

Reply:
yes this could break some test because that approach remove from the object if already existed a title
```ts
{title: 'hello', ...getFieldSchema({}, {setTitle: false})}
// returns
{title: undefined}
//instead of
{title: 'hello'}
```

Reply:
Interesting. Was the original version triggering this problem too? Do we have a test case to ensure that `{title: 'hello', ...getFieldSchema({}, {setTitle: false})}` works as intended?

Anyhow, if your proposed version works, then I am ok with it, but please do consider to add a test to prevent unintended regression in the future (if we don't have one already).